Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,267 @@
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+/// An enum that describes a parsed type in a cannonical form.
+public enum TypeDescription: Codable, Equatable, CustomStringConvertible {
+  /// A root type. e.g. Int
+  case simple(name: String)
+  /// A nested type. e.g. Array.Element
+  indirect case nested(name: String, parentType: TypeDescription)
+  /// A composed type. e.g. Identifiable & Equatable
+  indirect case composition([TypeDescription])
+  /// An optional type. e.g. Int?
+  indirect case optional(TypeDescription)
+  /// An implicitly unwrapped optional type. eg. Int!
+  indirect case implicitlyUnwrappedOptional(TypeDescription)
+  /// An array. e.g. [Int]
+  indirect case array(TypeDescription)
+  /// A dictionary. e.g. [Int: String]
+  indirect case dictionary(key: TypeDescription, value: TypeDescription)
+  /// A tuple. e.g. (Int, String)
+  indirect case tuple([TypeDescription])
+  /// A type that can't be represented by the above cases.
+  case unknown(text: String)
+
+  /*
+   * Note that we do not yet support the following syntax types:
+   * SomeTypeSyntax
+   * MetatypeTypeSyntax
+   * FunctionTypeSyntax
+   * AttributedTypeSyntax
+   * UnknownTypeSyntax
+   *
+   * We will likely need to add these types at some point in the future.
+   * We’ll Get There™
+   */
+
+  /// A description of this type that can be used for code-generation to represent this type.
+  public var description: String {
+    switch self {
+    case let .simple(name):
+      return name
+    case let .array(type):
+      return "[\(type)]"
+    case let .composition(types):
+      return types.map { $0.description }.joined(separator: " & ")
+    case let .optional(type):
+      return "\(type)?"
+    case let .implicitlyUnwrappedOptional(type):
+      return "\(type)!"
+    case let .dictionary(key, value):
+      return "[\(key): \(value)]"
+    case let .nested(name, parentType):
+      return "\(parentType).\(name)"
+    case let .tuple(types):
+      return "(\(types.map { $0.description }.joined(separator: ", ")))"
+    case let .unknown(text):
+      return text
+    }
+  }
+
+  public init(from decoder: Decoder) throws {
+    let values = try decoder.container(keyedBy: CodingKeys.self)
+    let caseDescription = try values.decode(String.self, forKey: .caseDescription)
+    if caseDescription == Self.simpleDescription {
```

Comment:
To avoid creating more conflicts for myself I'll waive off this until a later date since this feedback is purely style and doesn't affect API or function.

Reply:
Done in e3e854089058eeb9ec1cb921c1a898eefdbaf918 in https://github.com/fdiaz/SwiftInspector/pull/71/